samples/control_flow/never_return.jakt

```diff
@@ -0,0 +1,13 @@
+/// Expect: selfhost-only
+/// - output: ""
+
+function test() -> String {
+    if false {
+        return "hi"
+    }
+    abort()
+}
+
+function main() -> c_int {
+    abort()
```

Comment:
Wait. The fact that this test passes means Jakttest has a bug. It should error since the test aborts. Although it also suggests that we may need a `- no-run` flag so it only tries compiling it.

Reply:
Or maybe Jakttest is OK as it is, and we shouldn't care about the exit code of the test.